rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -74,6 +76,202 @@ rmw_connextdds_initialize_participant_factory_qos(
   return RMW_RET_OK;
 }
 
+static
+rmw_ret_t rmw_connextdds_extend_initial_peer_list(
+  const rmw_peer_address_t * const static_peers,
+  const size_t static_peer_count,
+  struct DDS_StringSeq * const out)
+{
+  if (static_peer_count == 0) {
+    return RMW_RET_OK;
+  }
+
+  if (static_peers == nullptr) {
+    RMW_CONNEXT_LOG_ERROR_A_SET(
+      "received nullptr static_peers but a static_peer_count of %lu",
+      static_peer_count);
+    return RMW_RET_ERROR;
+  }
+
+  const auto initial_length = DDS_StringSeq_get_length(out);
+  const auto new_seq_length = initial_length + static_peer_count;
+  if (!DDS_StringSeq_ensure_length(out, new_seq_length, new_seq_length)) {
+    RMW_CONNEXT_LOG_ERROR_SET("failed to resize string sequence")
+    return RMW_RET_ERROR;
+  }
+  for (size_t s = 0; s < static_peer_count; ++s) {
+    const auto index = initial_length + s;
+    const char * peer = static_peers[s].peer_address;
+    char ** const element_ref = DDS_StringSeq_get_reference(out, index);
+    RMW_CONNEXT_ASSERT(nullptr != element_ref);
+    if (nullptr != *element_ref) {
+      /* If some strings are still hanging around this dead space in the seq,
+         then free their memory. */
+      DDS_String_free(*element_ref);
+    }
+    *element_ref = DDS_String_dup(peer);
+    if (nullptr == *element_ref) {
+      RMW_CONNEXT_LOG_ERROR_A_SET("failed to duplicate peer string: %s", peer);
+      return RMW_RET_ERROR;
+    }
+
+    RMW_CONNEXT_LOG_TRACE_A(
+      "inserted static peer: i=%d, peer='%s'",
+      index, peer);
+  }
+
+  return RMW_RET_OK;
+}
+
+static
+rmw_ret_t
+rmw_connextdds_initialize_discovery_options(
+  rmw_context_impl_t * const ctx,
+  DDS_DomainParticipantQos & dp_qos)
+{
+  const auto range = ctx->discovery_options->automatic_discovery_range;
+  switch (range) {
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_SYSTEM_DEFAULT:
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_SUBNET:
+      /* No action needed. This is the default discovery behavior for DDS */
+      break;
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET:
+      RMW_CONNEXT_LOG_ERROR_SET(
+        "automatic_discovery_range unexpectedly RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET");
+      return RMW_RET_ERROR;
+      break;
```

Comment:
The `break` here is unnecessary.

Also, we could skip this case in favor of `default which does the same thing, unless you prefer having an explicit message to detect `NOT_SET`.

Reply:
Yeah, that's true.

I think an explicit message for NOT_SET is good to have.

Reply:
[e3a5c8f](https://github.com/ros2/rmw_connextdds/pull/108/commits/e3a5c8fe74d2b4419fc89c345b0ef59d5e9513df)